### PR TITLE
MNT: make fig.colorbar(..., ax=INPUT) even more forgiving

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Follow on to #24408

Accept anything that passes `np.iterable`

## PR Checklist

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -1419,10 +1418,14 @@ def make_axes(parents, location=None, orientation=None, fraction=0.15,
...
     # use .flatten or .ravel as these copy the references rather than
     # reuse them, leading to a memory leak
-    if isinstance(parents, np.ndarray):
-        parents = list(parents.flat)
-    elif not isinstance(parents, collections_abc.Sequence):
+    if np.iterable(parents):
+        if isinstance(parents, np.ndarray):
+            parents = list(parents.flat)
+        else:
+            parents = list(parents)
```
Comment (rcomer) [inline lib/matplotlib/colorbar.py:1425]:
```suggestion
    if isinstance(parents, np.ndarray):
        parents = list(parents.flat)
    elif np.iterable(parents):
        parents = list(parents)
```
?

**Review by rcomer — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/colorbar.py [hunk 2]
```diff
@@ -1395,7 +1394,7 @@ def make_axes(parents, location=None, orientation=None, fraction=0.15,
 
     Parameters
     ----------
-    parents : `~.axes.Axes` or sequence or `numpy.ndarray` of `~.axes.Axes`
+    parents : `~.axes.Axes` or iterable of `~.axes.Axes`
```
Comment (rcomer) [inline lib/matplotlib/colorbar.py:1397]: Should this still include ndarray explicitly?  A 2D array can be passed (and gets flattened), and I’m not sure it counts as an iterable of Axes (rather an iterable of iterables of Axes…)

Sorry I missed that before.

Comment (tacaswell) [reply lib/matplotlib/colorbar.py:1397]: oh, yeah that is correct, putting it back.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Anybody can merge after CI pass.

**Pull request merged**